implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpMeterRegistry.java

```diff
@@ -0,0 +1,282 @@
+                    .addDataPoints(histogramDataPoint));
+            return metricBuilder.build();
+        }
+
+        // if percentiles configured, use summary
+        if (histogramSnapshot.percentileValues().length != 0) {
+            SummaryDataPoint.Builder summaryData = SummaryDataPoint.newBuilder()
+                    .addAllAttributes(getTagsForId(histogramSupport.getId()))
+                    .setTimeUnixNano(wallTimeNanos)
+                    .setSum(histogramSnapshot.total())
+                    .setCount(histogramSnapshot.count());
+            for (ValueAtPercentile percentile : histogramSnapshot.percentileValues()) {
+                summaryData.addQuantileValues(SummaryDataPoint.ValueAtQuantile.newBuilder()
+                        .setQuantile(percentile.percentile()).setValue(percentile.value()));
+            }
+            metricBuilder.setSummary(Summary.newBuilder().addDataPoints(summaryData));
+            return metricBuilder.build();
+        }
+
+        return metricBuilder.setHistogram(Histogram.newBuilder()
+                .setAggregationTemporality(AggregationTemporality.AGGREGATION_TEMPORALITY_CUMULATIVE)
+                .addDataPoints(histogramDataPoint)).build();
+    }
+
+    private Metric writeFunctionTimer(FunctionTimer functionTimer) {
+        return getMetricBuilder(functionTimer.getId())
+                .setHistogram(Histogram.newBuilder()
+                        .addDataPoints(HistogramDataPoint.newBuilder()
+                                .addAllAttributes(getTagsForId(functionTimer.getId()))
+                                // TODO start time
+                                .setTimeUnixNano(TimeUnit.MILLISECONDS.toNanos(this.clock.wallTime()))
+                                .setSum(functionTimer.totalTime(getBaseTimeUnit()))
+                                .setCount((long) functionTimer.count())))
+                .build();
+    }
+
+    private Metric.Builder getMetricBuilder(Meter.Id id) {
+        Metric.Builder builder = Metric.newBuilder()
+                .setName(getConventionName(id));
+        if (id.getBaseUnit() != null) {
+            builder.setUnit(id.getBaseUnit());
+        }
+        if (id.getDescription() != null) {
+            builder.setDescription(id.getDescription());
+        }
+        return builder;
+    }
+
+    private Iterable<? extends KeyValue> getTagsForId(Meter.Id id) {
+        return id.getTags().stream()
+                .map(tag -> KeyValue.newBuilder()
+                        .setKey(tag.getKey())
+                        .setValue(AnyValue.newBuilder().setStringValue(tag.getValue()).build())
+                        .build())
+                .collect(Collectors.toList());
+    }
+
+    private Iterable<KeyValue> getResourceAttributes() {
+        List<KeyValue> attributes = new ArrayList<>();
+        // TODO How to expose configuration of the service.name
```

Comment:
In, for example, the DatadogMeterRegistry, we take configuration for the tag name to convert to the Datadog-specified `host` tag. This allows you to configure a common tag that will be published to all registries, but will be converted to what Datadog expects.

I thought of doing something similar here, but it's a bit more tricky. We're only expecting one Resource, I think. And then the question is, which meter to look at to get the tag that corresponds to the Resource `service.name`? And when?

A configuration option to set this directly instead of taking it from a configured tag is another option, which we have done in some registries. But this can result in this `service.name` info only being in your OTLP metrics, while missing from meters in other registries, which is generally not desired.